remote/gitea/gitea.go

```diff
@@ -0,0 +1,344 @@
+package gitea
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/drone/drone/model"
+	"github.com/drone/drone/remote"
+
+	"code.gitea.io/sdk/gitea"
+)
+
+// Opts defines configuration options.
+type Opts struct {
+	URL         string // Gitea server url.
+	Username    string // Optional machine account username.
+	Password    string // Optional machine account password.
+	PrivateMode bool   // Gitea is running in private mode.
+	SkipVerify  bool   // Skip ssl verification.
+}
+
+type client struct {
+	URL         string
+	Machine     string
+	Username    string
+	Password    string
+	PrivateMode bool
+	SkipVerify  bool
+}
+
+const (
+	DescPending  = "the build is pending"
+	DescRunning  = "the build is running"
+	DescSuccess  = "the build was successful"
+	DescFailure  = "the build failed"
+	DescCanceled = "the build canceled"
+	DescBlocked  = "the build is pending approval"
+	DescDeclined = "the build was rejected"
+)
+
+// getStatus is a helper functin that converts a Drone
+// status to a GitHub status.
+func getStatus(status string) gitea.StatusState {
+	switch status {
+	case model.StatusPending, model.StatusBlocked:
+		return gitea.StatusPending
+	case model.StatusRunning:
+		return gitea.StatusPending
+	case model.StatusSuccess:
+		return gitea.StatusSuccess
+	case model.StatusFailure, model.StatusError:
+		return gitea.StatusFailure
+	case model.StatusKilled:
+		return gitea.StatusFailure
+	default:
```

Comment:
This is related to the gating of drone if the gating have been declined. So I think gitea.StatusWarning makes sense.